tx/textual/valuerenderer/coins.go

```diff
@@ -76,7 +77,42 @@ func (vr coinsValueRenderer) Format(ctx context.Context, v protoreflect.Value) (
 	}
 }
 
-func (vr coinsValueRenderer) Parse(_ context.Context, screens []Screen) (protoreflect.Value, error) {
-	// ref: https://github.com/cosmos/cosmos-sdk/issues/13153
-	panic("implement me, see #13153")
+func (vr coinsValueRenderer) Parse(ctx context.Context, screens []Screen) (protoreflect.Value, error) {
+	if len(screens) != 1 {
+		return protoreflect.Value{}, fmt.Errorf("expected single screen: %v", screens)
+	}
+
+	coins := strings.Split(screens[0].Text, ", ")
+	metadatas := make([]*bankv1beta1.Metadata, len(coins))
+
+	if len(coins) > 1 {
+		var err error
+		for i, coin := range coins {
+			coinArr := strings.Split(coin, " ")
+			metadatas[i], err = vr.coinMetadataQuerier(ctx, coinArr[1])
+			if err != nil {
+				return protoreflect.Value{}, err
+			}
+		}
+
+		parsed, err := corecoins.ParseCoins(coins, metadatas)
+		if err != nil {
+			return protoreflect.Value{}, err
+		}
+
+		return protoreflect.ValueOf(NewGenericList(parsed)), err
+	} else {
```

Comment:
I'm not sure we need an if/else here. We can do simply do a `range coins`, and it will work in both cases len(coins) == 1 and > 1.